code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1147 @@
+	icon_state = "rage_rune"
+	///fire overlay appearing behind someone when they are on the rune
+	var/mutable_appearance/hate_fire
+
+
+/obj/effect/warped_rune/redspace/Initialize()
+	. = ..()
+	hate_fire = image('icons/mob/OnFire.dmi',"Standing", RAGE_FIRE_LAYER)
+	hate_fire.color = "#C01414" //dark red kind of.
+	hate_fire.layer = RAGE_FIRE_LAYER
+
+///boost up the unarmed damage of the person currently on the tile.
+/obj/effect/warped_rune/redspace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(!istype(crossing,/mob/living/carbon/human))
+		return
+
+	var/mob/living/carbon/human/enraged = crossing
+	enraged.add_overlay(hate_fire)
+	RegisterSignal(enraged, COMSIG_HUMAN_EARLY_UNARMED_ATTACK, .proc/genetic_punch)
+	to_chat(enraged, "<span class='warning'>You feel the hate of a thousand souls culminating into your fists!</span>")
+
+
+/obj/effect/warped_rune/redspace/proc/genetic_punch(mob/living/puncher, atom/attacked)
+	if(!istype(attacked,/mob/living/carbon/human) || puncher.a_intent != INTENT_HARM)
+		return
+
+	var/mob/living/carbon/human/punched = attacked
+	if(punched.mind && punched.stat != DEAD)
+		puncher.adjustBruteLoss(-20)//Heal the puncher per punch. Only works on players that are alive.
+		puncher.adjustFireLoss(-20)
+	punched.adjustCloneLoss(20) //genetic damage is the closest thing we have to soul damage
+	playsound(puncher, 'sound/effects/pop_expl.ogg', 75, TRUE)
+	to_chat(puncher, "<span class='warning'>You punch [punched]'s soul with all your might!</span>")
+	to_chat(punched, "<span class='warning'>You feel like something very important got punched out of you!</span>")
+	puncher.do_attack_animation(punched, ATTACK_EFFECT_SMASH)
+	return COMPONENT_NO_ATTACK_HAND //here to make sure we replace the original punch with our genetic punch
+
+
+///takes away the overlay and the genetic punch when you leave the rune
+/obj/effect/warped_rune/redspace/Uncrossed(atom/movable/crossing)
+	. = ..()
+	if(!istype(crossing,/mob/living/carbon/human))
+		return
+	var/mob/living/carbon/human/enraged = crossing
+	UnregisterSignal(enraged, COMSIG_HUMAN_EARLY_UNARMED_ATTACK)
+	enraged.cut_overlay(hate_fire)
+
+///destroying the rune will also remove the punch force of the persons on the rune.
+/obj/effect/warped_rune/redspace/Destroy()
+	for(var/mob/living/carbon/human/enraged in rune_turf)
+		UnregisterSignal(enraged, COMSIG_HUMAN_EARLY_UNARMED_ATTACK)
+		enraged.cut_overlay(hate_fire)
+	return ..()
+
+
+/* Green rune vores plasma and spews out xeno resin which lets you build xeno structure*/
+
+
+/obj/item/slimecross/warping/green
+	colour = "green"
+	runepath = /obj/effect/warped_rune/greenspace
+	drawing_time = 100
+
+
+/obj/effect/warped_rune/greenspace
+	icon_state = "xeno_rune"
+	desc = "We will build walls out of our fallen foes, they shall fear our very buildings."
+	max_cooldown = 100
+	///path of the xeno resin sheet to spawn by the rune
+	var/obj/item/stack/sheet/xeno_resin/resin
+
+
+/obj/item/stack/sheet/xeno_resin
```

Comment:
it'd be nice if you could make it so that (real) xenomorphs can hold these resin sheets, sort of like how they can hold facehuggers and the like